test/test_custom_ops.py

```diff
@@ -1730,6 +1730,28 @@ def test_is_functional_schema(self):
             res = torch._library.utils.is_functional_schema(schema)
             self.assertEqual(res, expected)
 
+    def test_is_tensorlist_like_type(self):
+        tensorlists = [
+            # Tensor[]
+            torch.ops.aten.where.default._schema.returns[0].type,
+            # Tensor?[]
+            torch.ops.aten.index.Tensor._schema.arguments[1].type,
+            # Tensor[]?
+            torch._C.parse_schema("foo(Tensor[]? x) -> ()").arguments[0].type,
+            # Tensor?[]?
```

Comment:
Are these actually valid schemas??

Reply:
It's unclear to me?

Reply:
I'm surprised that parse_schema doesn't raise....

Reply:
Yes, these are all valid (and used)